webapp/channels/src/components/profile_popover/profile_popover_controller.tsx

```diff
@@ -0,0 +1,163 @@
+     * If not provided, the popover will automatically determine the previously focused element
+     * and focus that on close. However, if the previously focused element is not correctly detected
+     * by the popover, or the previously focused element will disappear after the popover opens,
+     * it is necessary to provide this function to focus the correct element.
+     */
+    returnFocus?: () => void;
+
+    onToggle?: (isMounted: boolean) => void;
+}
+
+export function ProfilePopoverController<TriggerComponentType = HTMLSpanElement>(props: Props<TriggerComponentType>) {
+    const [isOpen, setOpen] = useState(false);
+
+    const {refs, floatingStyles, context: floatingContext} = useFloating({
+        open: isOpen,
+        onOpenChange: setOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [autoPlacement()],
+    });
+
+    const {isMounted, styles: transitionStyles} = useTransitionStyles(floatingContext, {
+        duration: {
+            open: PROFILE_POPOVER_OPENING_DELAY,
+            close: PROFILE_POPOVER_CLOSING_DELAY,
+        },
+    });
+
+    const combinedFloatingStyles = Object.assign({}, floatingStyles, transitionStyles);
+
+    const clickInteractions = useClick(floatingContext);
+
+    const dismissInteraction = useDismiss(floatingContext);
+
+    const role = useRole(floatingContext);
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([
+        clickInteractions,
+        dismissInteraction,
+        role,
+    ]);
+
+    const handleHide = useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    const TriggerComponent = props.triggerComponentAs ?? 'span';
+
+    return (
+        <>
+            <TriggerComponent
+                id={props.triggerComponentId}
+                ref={refs.setReference}
+                className={props.triggerComponentClass}
+                style={props.triggerComponentStyle}
+                {...getReferenceProps()}
+            >
+                {props.children}
+            </TriggerComponent>
+
+            {isMounted && (
+                <FloatingPortal id='user-profile-popover-portal'>
+                    <FloatingOverlay
+                        id='user-profile-popover-floating-overlay'
+                        className='user-profile-popover-floating-overlay'
+                        lockScroll={true}
+                    >
+                        <FloatingFocusManager context={floatingContext}>
+                            <div
+                                ref={refs.setFloating}
+                                style={combinedFloatingStyles}
+                                className={classNames('user-profile-popover', A11yClassNames.POPUP)}
+                                {...getFloatingProps()}
+                                id='user-profile-popover' // This is placed here as id is being overwritten by getFloatingProps
```

Comment:
Are you sure that it's okay to overwrite this? I'd assume that Floating UI wants us to use whatever ID that it's giving us because it probably uses that ID to find the floating element

Reply:
Yes i think its safer to remove it